pkg/buildkit/worker/pool.go

```diff
@@ -502,8 +520,44 @@ func (p *workerPool) triggerReconcile() {
 	}
 }
 
+// ensure pod is operational by checking its phase and conditions
+func (p *AutoscalingPool) isOperationalPod(ctx context.Context, podName string) (verdict bool) {
+	// fetch the latest version of the pod
+	pod, err := p.podClient.Get(ctx, podName, metav1.GetOptions{})
```

Comment:
pretty minor, but should this have a resource version associated with the request (i.e. no older than the one we got from the list)?

Reply:
maybe? i look at this again, thanks for raising it.

Reply:
@steved https://kubernetes.io/docs/reference/using-api/api-concepts/#semantics-for-get-and-list

> For **get**, the semantics of resourceVersion are: `resourceVersion` unset == "Most Recent"

Reply:
Makes sense, I was just curious if we should be explicit. Thanks for checking.